### PR TITLE
[FLINK-11614] [docs-zh] Translate the "Configuring Dependencies" page into Chinese

### PR DESCRIPTION
**Pull request by Leeviiii**
…iguring Dependencies" page into Chinese

## What is the purpose of the change

This pull request completes the Chinese translation of "Configuring Dependencies"  page from official document.

## Brief change log

*Translate the "Configuring Dependencies" page into Chinese*
  
## Verifying this change

*This change is to add a new translated document.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive):  no 
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no 
  - The S3 file system connector:  no 

## Documentation

  - Does this pull request introduce a new feature?  no

### CODE REVIEW

**Comment by klion26** [comment]
@Leeviiii thanks for your contribution, we'll wait for #7754 to be merged, so we can easily know what the changes are.

**Comment by Leeviiii** [comment]
@klion26 ok, thanks

**Comment by bowenli86** [comment]
This is some really good stuff! Exciting to see some official users' contribution on doc translation. Can you retitle the PR to be complete?

BTW, we may also need to wait for https://github.com/apache/flink-web/pull/190 on finalizing the review process.

**Comment by wuchong** [comment]
Thanks @Leeviiii ,  #7754 and  apache/flink-web#190 has been merged. So I think we can start to review this PR.

Before that, can you rebase your commits to the latest master ?

**Comment by Leeviiii** [comment]
first I 'git rebase -i master'
then recommit my PR which make some changes after reading the  “Flink Translation Specifications”.
But it seems that there is something wrong.There are 580 files changed ,but I only change one.
I do not know why?

**Comment by wuchong** [comment]
Hi @Leeviiii ,  please do not MERGE master. You can use `IDEA > VCS > Git > Rebase` tool to rebase commits if you are using IntelliJ IDEA. 

You can also create a new clean branch, and copy the changes to the new branch, and then force update your remote Leeviiii:FLINK-11614 branch.

**Comment by Leeviiii** [comment]
ok @wuchong  I have clean the branch as what you said.

**Comment by wuchong** [comment]
Thanks @Leeviiii , it looks good now. I will find time to review it then.

File: docs/dev/projectsetup/dependencies.zh.md [hunk 1]
```diff
@@ -22,47 +22,36 @@ specific language governing permissions and limitations
...
-    The user application is typically packaged into an *application jar*, which contains the application code and the required
-    connector and library dependencies.
+    用户应用程序依赖项不需包括 Flink DataSet / DataStream API 以及运行时依赖项，因为它们已经是 Flink 核心依赖项的一部分。
 
-    The user application dependencies explicitly do not include the Flink DataSet / DataStream APIs and runtime dependencies,
-    because those are already part of Flink's Core Dependencies.
+## 搭建一个项目: 基础依赖
 
+开发 Flink 应用程序需要最低限度的 API 依赖。Maven 用户，可以使用 
+[Java 项目模板]({{ site.baseurl }}/dev/projectsetup/java_api_quickstart.zh.html)或者
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
[Java 项目模板]({{ site.baseurl }}/zh/dev/projectsetup/java_api_quickstart.html)或者
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 2]
```diff
@@ -22,47 +22,36 @@ specific language governing permissions and limitations
...
+[Scala 项目模板]({{ site.baseurl }}/dev/projectsetup/scala_api_quickstart.zh.html)来创建一个包含最初依赖的程序骨架。
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
[Scala 项目模板]({{ site.baseurl }}/zh/dev/projectsetup/scala_api_quickstart.html)来创建一个包含最初依赖的程序骨架。
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 3]
```diff
@@ -22,47 +22,36 @@ specific language governing permissions and limitations
...
-## Setting up a Project: Basic Dependencies
-
-Every Flink application needs as the bare minimum the API dependencies, to develop against.
-For Maven, you can use the [Java Project Template]({{ site.baseurl }}/dev/projectsetup/java_api_quickstart.html)
-or [Scala Project Template]({{ site.baseurl }}/dev/projectsetup/scala_api_quickstart.html) to create
-a program skeleton with these initial dependencies.
-
-When setting up a project manually, you need to add the following dependencies for the Java/Scala API
-(here presented in Maven syntax, but the same dependencies apply to other build tools (Gradle, SBT, etc.) as well.
+手动设置项目时，需要为Java / Scala API 添加以下依赖项(这里以 Maven 语法为例，但也适用于其他构建工具( Gradle、 SBT 等))。
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
手动设置项目时，需要为 Java 或 Scala API 添加以下依赖项（这里以 Maven 语法为例，但也适用于其他构建工具（Gradle、 SBT 等））。
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 4]
```diff
@@ -99,31 +88,23 @@ When setting up a project manually, you need to add the following dependencies f
...
-which are already available in any setup.
+**注意事项:** 所有这些依赖项的作用域都应该设置为 *provided* 。
+这意味着需要这些依赖进行编译，但不应将它们打包到项目生成的应用程序jar文件中-- 
+因为这些依赖项是 Flink 的核心依赖，在应用启动前已经是可用的状态了。
 
-It is highly recommended to keep the dependencies in scope *provided*. If they are not set to *provided*,
-the best case is that the resulting JAR becomes excessively large, because it also contains all Flink core
-dependencies. The worst case is that the Flink core dependencies that are added to the application's jar file
-clash with some of your own dependency versions (which is normally avoided through inverted classloading).
+我们强烈建议保持这些依赖的作用域为 *provided* 。 如果它们的作用域未设置为 *provided* ，则最好的情况是因为包含了 Flink 的核心依赖而导致生成的jar包变得过大。
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
我们强烈建议保持这些依赖的作用域为 *provided* 。 如果它们的作用域未设置为 *provided* ，则典型的情况是因为包含了 Flink 的核心依赖而导致生成的jar包变得过大。
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 5]
```diff
@@ -99,31 +88,23 @@ When setting up a project manually, you need to add the following dependencies f
...
-**Note on IntelliJ:** To make the applications run within IntelliJ IDEA, the Flink dependencies need
-to be declared in scope *compile* rather than *provided*. Otherwise IntelliJ will not add them to the classpath and
-the in-IDE execution will fail with a `NoClassDefFountError`. To avoid having to declare the
-dependency scope as *compile* (which is not recommended, see above), the above linked Java- and Scala
-project templates use a trick: They add a profile that selectively activates when the application
-is run in IntelliJ and only then promotes the dependencies to scope *compile*, without affecting
-the packaging of the JAR files.
+**IntelliJ 上的一些注意事项:** 为了可以让 Flink 应用在 IntelliJ IDEA 中运行，这些 Flink 核心依赖的作用域需要设置为 *compile* 而不是 *provided* 。
+否则 IntelliJ 不会添加这些依赖到 classpath，会导致应用运行时抛出 `NoClassDefFountError` 异常。为了避免声明这些依赖的作用域为 *compile* (因为我们不推荐这样做)，
+上文给出的的 Java 和 Scala 项目模板使用了一个小技巧：添加了一个可选择激活配置，仅当应用程序在 IntelliJ 中运行时该配置被激活，
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
上文给出的 Java 和 Scala 项目模板使用了一个小技巧：添加了一个 profile，仅当应用程序在 IntelliJ 中运行时该 profile 会被激活，
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 6]
```diff
@@ -99,31 +88,23 @@ When setting up a project manually, you need to add the following dependencies f
...
+然后将依赖作用域设置为 *compile* ，从而不影响应用 jar 包。
 
+## 添加 Connectors 以及类库依赖
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]: Connectors -> 连接器

File: docs/dev/projectsetup/dependencies.zh.md [hunk 7]
```diff
@@ -99,31 +88,23 @@ When setting up a project manually, you need to add the following dependencies f
...
 
-## Adding Connector and Library Dependencies
+大多数应用需要依赖特定的 Connectors 或其他类库，例如 Kafka、Cassandra 的 Connectors 等。这些 Connectors 不是 Flink 核心依赖的一部分，因此必须作为依赖项手动添加到应用程序中。
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
大多数应用需要依赖特定的连接器或其他类库，例如 Kafka、Cassandra 的连接器等。这些连接器不是 Flink 核心依赖的一部分，因此必须作为依赖项手动添加到应用程序中。
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 8]
```diff
@@ -99,31 +88,23 @@ When setting up a project manually, you need to add the following dependencies f
...
 
-Most applications need specific connectors or libraries to run, for example a connector to Kafka, Cassandra, etc.
-These connectors are not part of Flink's core dependencies and must hence be added as dependencies to the application
-
-Below is an example adding the connector for Kafka 0.10 as a dependency (Maven syntax):
+下面是添加 Kafka 0.10 Connectors 依赖（Maven 语法）的示例：
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
下面是添加 Kafka 0.10 连接器依赖（Maven 语法）的示例：
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 9]
```diff
@@ -132,63 +113,45 @@ Below is an example adding the connector for Kafka 0.10 as a dependency (Maven s
...
-
-Projects created from the [Java Project Template]({{ site.baseurl }}/dev/projectsetup/java_api_quickstart.html) or
-[Scala Project Template]({{ site.baseurl }}/dev/projectsetup/scala_api_quickstart.html) are configured to automatically include
-the application dependencies into the application jar when running `mvn clean package`. For projects that are
-not set up from those templates, we recommend to add the Maven Shade Plugin (as listed in the Appendix below)
-to build the application jar with all required dependencies.
+我们建议将应用程序代码及其所有需要的依赖项打包到一个 *jar-with-dependencies* 的 jar 包中。
+这个打包好的应用 jar 可以提交到已经运行的 Flink 集群中，或者添加到 Flink 应用容器镜像中。
+ 
+通过[Java 项目模板]({{ site.baseurl }}/dev/projectsetup/java_api_quickstart_zh.html) 或者
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
通过[Java 项目模板]({{ site.baseurl }}/zh/dev/projectsetup/java_api_quickstart.html) 或者
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 10]
```diff
@@ -132,63 +113,45 @@ Below is an example adding the connector for Kafka 0.10 as a dependency (Maven s
...
+[Scala 项目模板]({{ site.baseurl }}/dev/projectsetup/scala_api_quickstart_zh.html) 创建的应用，
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
[Scala 项目模板]({{ site.baseurl }}/zh/dev/projectsetup/scala_api_quickstart.html) 创建的应用，
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 11]
```diff
@@ -132,63 +113,45 @@ Below is an example adding the connector for Kafka 0.10 as a dependency (Maven s
...
-Scala 2.12.
+所有依赖 Scala 的 Flink 类库都以它们依赖的 Scala 版本为后缀，例如 `flink-streaming-scala_2.11`。
 
-All Flink dependencies that (transitively) depend on Scala are suffixed with the
-Scala version that they are built for, for example `flink-streaming-scala_2.11`.
+只使用 Java 的开发人员可以选择任何 Scala 版本，Scala 开发人员需要选择与其应用程序相匹配的 Scala 版本。
 
-Developers that only use Java can pick any Scala version, Scala developers need to
-pick the Scala version that matches their application's Scala version.
+对于指定的 Scala 版本如何构建 Flink 应用可以参考 [构建指南]({{ site.baseurl }}/flinkDev/building_zh.html#scala-versions)
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
对于指定的 Scala 版本如何构建 Flink 应用可以参考 [构建指南]({{ site.baseurl }}/zh/flinkDev/building.html#scala-versions) 
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 12]
```diff
@@ -132,63 +113,45 @@ Below is an example adding the connector for Kafka 0.10 as a dependency (Maven s
...
+## Hadoop 版本
 
-## Hadoop Dependencies
+**一般规则：永远不需要将 Hadoop 依赖项直接添加到你的应用程序中** 
+*(唯一的例外是使用 Flink 的 Hadoop 兼容包装器来处理 Hadoop 格式的输入/输出时)*
 
-**General rule: It should never be necessary to add Hadoop dependencies directly to your application.**
-*(The only exception being when using existing Hadoop input-/output formats with Flink's Hadoop compatibility wrappers)*
+如果你想要在 Flink 应用中使用 Hadoop，你需要使用包含 Hadoop 依赖的 Flink，而非将 Hadoop 作为应用依赖进行添加。
+请参考[Hadoop 构建指南]({{ site.baseurl }}/ops/deployment/hadoop_zh.html)
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
请参考[Hadoop 构建指南]({{ site.baseurl }}/zh/ops/deployment/hadoop.html)
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 13]
```diff
@@ -213,8 +176,8 @@ you can use the following shade plugin definition:
                         </artifactSet>
                         <filters>
                             <filter>
-                                <!-- Do not copy the signatures in the META-INF folder.
-                                Otherwise, this might cause SecurityExceptions when using the JAR. -->
+                                <!--不要拷贝 META-INF 目录下的前面，
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
                                <!--不要拷贝 META-INF 目录下的签名，
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 14]
```diff
@@ -132,63 +113,45 @@ Below is an example adding the connector for Kafka 0.10 as a dependency (Maven s
...
 
-If you want to use Flink with Hadoop, you need to have a Flink setup that includes the Hadoop dependencies, rather than
-adding Hadoop as an application dependency. Please refer to the [Hadoop Setup Guide]({{ site.baseurl }}/ops/deployment/hadoop.html)
-for details.
+这样设计是出于两个主要原因:
+  - 可能在用户程序启动之前， 一些 Hadoop 操作就已经发生在 Flink 核心中，比如为 checkpoint 设置 HDFS 路径，通过 Hadoop's Kerberos tokens 进行权限认证以及进行 YARN 部署等。
 
-There are two main reasons for that design:
+  - Flink 的反向类加载方法隐藏了核心依赖关系中的许多传递依赖关系。这不仅适用于 Flink 自己的核心依赖项，也适用于 Hadoop 在启动中存在的依赖项。
+    通过这种方式，应用程序可以使用相同依赖项的不同版本，而不会引起依赖项的冲突（若非若此可能会引起严重依赖问题，因为 Hadoops 依赖树十分庞大。）
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
    通过这种方式，应用程序可以使用相同依赖项的不同版本，而不会引起依赖项的冲突（若非如此可能会引起严重依赖问题，因为 Hadoops 依赖树十分庞大。）
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 15]
```diff
@@ -132,63 +113,45 @@ Below is an example adding the connector for Kafka 0.10 as a dependency (Maven s
...
-**General rule: It should never be necessary to add Hadoop dependencies directly to your application.**
-*(The only exception being when using existing Hadoop input-/output formats with Flink's Hadoop compatibility wrappers)*
+如果你想要在 Flink 应用中使用 Hadoop，你需要使用包含 Hadoop 依赖的 Flink，而非将 Hadoop 作为应用依赖进行添加。
+请参考[Hadoop 构建指南]({{ site.baseurl }}/ops/deployment/hadoop_zh.html)
 
-If you want to use Flink with Hadoop, you need to have a Flink setup that includes the Hadoop dependencies, rather than
-adding Hadoop as an application dependency. Please refer to the [Hadoop Setup Guide]({{ site.baseurl }}/ops/deployment/hadoop.html)
-for details.
+这样设计是出于两个主要原因:
+  - 可能在用户程序启动之前， 一些 Hadoop 操作就已经发生在 Flink 核心中，比如为 checkpoint 设置 HDFS 路径，通过 Hadoop's Kerberos tokens 进行权限认证以及进行 YARN 部署等。
```
Comment (wuchong) [inline docs/dev/projectsetup/dependencies.zh.md]:
```suggestion
  - 可能在用户程序启动之前，一些 Hadoop 交互操作就已经发生在 Flink 核心中了，比如为 checkpoint 设置 HDFS 路径，通过 Hadoop's Kerberos tokens 进行权限认证以及进行 YARN 部署等。
```

File: docs/dev/projectsetup/dependencies.zh.md [hunk 16]
```diff
@@ -22,47 +22,36 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-Every Flink application depends on a set of Flink libraries. At the bare minimum, the application depends
-on the Flink APIs. Many applications depend in addition on certain connector libraries (like Kafka, Cassandra, etc.).
-When running Flink applications (either in a distributed deployment, or in the IDE for testing), the Flink
-runtime library must be available as well.
+每个 Flink 应用都需要依赖一组 Flink 类库。Flink 应用至少需要依赖 Flink APIs。许多应用还会额外的依赖 Connectors (比如 Kafka、Cassandra 等)。
```
[the remark this answers is not in the document]
Comment (Leeviiii) [reply docs/dev/projectsetup/dependencies.zh.md]: ok , I will change that

File: docs/dev/projectsetup/dependencies.zh.md (hunk 5 above)
Comment (Leeviiii) [reply docs/dev/projectsetup/dependencies.zh.md]: I think "profile 会被激活" -> "profile 才会被激活" will be more proper

**Comment by Leeviiii** [comment]
@wuchong  I have fxied by what you  suggesstion said

File: docs/dev/projectsetup/dependencies.zh.md (hunk 5 above)
Comment (wuchong) [reply docs/dev/projectsetup/dependencies.zh.md]: Yes, agree with you.

**Comment by Leeviiii** [comment]
@wuchong   I am sorry to disturb you .  I had fxied by what you suggesstion said, so can you help me review again? thanks

**Comment by wuchong** [comment]
Thanks @Leeviiii for the contribution and sorry for the late response. 

The PR looks good to me, I will merge this soon.

@flinkbot approve all.

**Pull request closed without merging**